jfr-connection/src/test/java/io/opentelemetry/contrib/jfr/connection/RecordingTest.java

```diff
@@ -505,4 +513,43 @@ void assertRecordingCloneState() {
       fail("Error thrown by MBean server or FlightRecorderMXBean: ", badBean);
     }
   }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"5 s", "5m"})
+  void assertDiagnosticCommandCanRecordWithDuration(String duration) {
+    // Issue #1338
+    MBeanServerConnection mBeanServer = ManagementFactory.getPlatformMBeanServer();
+    try {
+      FlightRecorderConnection connection =
+          FlightRecorderDiagnosticCommandConnection.connect(mBeanServer);
+      assertCanRecord(connection, duration);
+    } catch (IOException | JfrConnectionException e) {
+      fail(e);
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"5 s", "5m"})
+  void assertMbeanCanRecordWithDuration(String duration) {
+    // Ensure fix for issue #1338 doesn't regress MBean recording
+    MBeanServerConnection mBeanServer = ManagementFactory.getPlatformMBeanServer();
+    try {
+      FlightRecorderConnection connection = FlightRecorderConnection.connect(mBeanServer);
+      assertCanRecord(connection, duration);
+    } catch (IOException | JfrConnectionException e) {
+      fail(e);
+    }
+  }
+
+  private static void assertCanRecord(FlightRecorderConnection connection, String duration)
+      throws JfrConnectionException {
+    RecordingOptions recordingOptions = new RecordingOptions.Builder().duration(duration).build();
+    RecordingConfiguration recordingConfiguration = RecordingConfiguration.PROFILE_CONFIGURATION;
+    try (Recording recording = connection.newRecording(recordingOptions, recordingConfiguration)) {
+      recording.start();
```

Comment:
What do you think about dumping the recording and verifying that the JFR file exists or contains events? It may be a stronger verification to check that the things have worked fine.

Reply:
It's somewhat non-determinstic when the recording will actually start. And if there is some other test that starts a recording, then those recorded events might also be included it the check. I guess you could filter the events by recording id, though.

Reply:
The tests are executed in a sequential way. The JFR file of another test execution can be removed at the beginning of the current test, as it's done in https://github.com/open-telemetry/opentelemetry-java-contrib/blob/7b32097367b03bebd18c8131a041739c9212f158/jfr-connection/src/test/java/io/opentelemetry/contrib/jfr/connection/RecordingConfigurationTest.java

Reply:
For this test, I really only care that the diagnostic command and the MBean don't throw an exeception. I think adding a test to verify the recording works is a different test. I'd argue that that kind of test should be done in a separate PR since it could be complex and I'd hate to hold up this PR on account of it.